packages/hint-compat-api/src/helpers/compat-api.ts

```diff
@@ -52,34 +51,31 @@ export class CompatApi {
     }
 
     private getFeatureByBrowsers(featureValue: CompatStatement & MDNTreeFilteredByBrowsers): CompatStatement | null {
-        const typedFeatures = {} as CompatStatement & MDNTreeFilteredByBrowsers;
-
-        typedFeatures.__compat = featureValue.__compat;
+        const childrenFeatures = this.getFeaturesAndChildrenRequiredToTest(featureValue);
+        const hasChildrenFeatures = Object.keys(childrenFeatures).length > 0;
+        const typedFeatures = { ...childrenFeatures, __compat: featureValue.__compat } as CompatStatement & MDNTreeFilteredByBrowsers;
 
-        const isChildRequired = this.getFeaturesAndChildrenRequiredToTest(typedFeatures, featureValue);
-
-        if (!isChildRequired && !this.isFeatureRequiredToTest(featureValue)) {
+        if (!hasChildrenFeatures && !this.isFeatureRequiredToTest(featureValue)) {
             return null;
         }
 
         return typedFeatures;
     }
 
-    private getFeaturesAndChildrenRequiredToTest(typedFeatures: CompatStatement & MDNTreeFilteredByBrowsers, featureValue: CompatStatement & MDNTreeFilteredByBrowsers): boolean {
-        if (typeof featureValue === 'object' && Object.keys(featureValue).length > 1) {
-            return Object.entries(featureValue as object).some(([childKey, childValue]) => {
-
-                if (!this.isFeatureRequiredToTest(childValue as CompatStatement & MDNTreeFilteredByBrowsers)) {
-                    return false;
-                }
-
-                typedFeatures[childKey] = childValue;
+    private getFeaturesAndChildrenRequiredToTest(featureValue: CompatStatement & MDNTreeFilteredByBrowsers): CompatStatement {
+        const typedFeatures = {} as CompatStatement & MDNTreeFilteredByBrowsers;
 
-                return true;
-            });
+        if (typeof featureValue === 'object' && Object.keys(featureValue).length) {
+            Object.entries(featureValue as object)
```

Comment:
Why `filter` and then `forEach`? In the worst case, you are iterating twice the same collection.
Why don't do everything in the `forEach`?